### PR TITLE
feat(nemotron): allow skipping logits materialization

### PR DESCRIPTION
**Pull request by pzelasko**
## What

- Add an opt-in `compute_logits=False` argument to `NemotronHForCausalLM.forward`.
- Skip the LM-head projection while preserving final hidden states for fused linear cross entropy.
- Reject unsupported use with labels or pipeline parallelism; default behavior is unchanged.

## Why

Fused linear cross entropy consumes final hidden states and the LM-head weight directly. Materializing the full vocabulary logits first wastes substantial peak memory for long packed speech sequences. This keeps the capability generic in Automodel and leaves SALM-specific loss routing in NeMo.

MTP execution is unchanged and remains enabled.

## Impact

The new path is opt-in. Existing training, inference, MTP, and pipeline-parallel callers retain the current behavior.

## Checks

- `pytest -q tests/unit_tests/models/nemotron_v3/test_nemotron_v3_mtp.py -k compute_logits_false`
- `python -m py_compile nemo_automodel/components/models/nemotron_v3/model.py`
- `git diff --check`

The corresponding NeMo integration was also validated in a 128-step, 8xH100 packed-speech gate with MTP retained.

### CODE REVIEW

**Comment by pzelasko** [comment]
/ok to test [a1df443](https://github.com/NVIDIA-NeMo/Automodel/pull/3555/commits/a1df443105b2dfc413b703f623e2529f3b1d6d82)

File: nemo_automodel/components/moe/megatron/token_dispatcher.py
Comment (HuiyingLi) [inline nemo_automodel/components/moe/megatron/token_dispatcher.py:534]: this might be from the prev PR. Could you please merge main &  resolve conflict? Thanks!

**Comment by pzelasko** [comment]
/ok to test [46ba282](https://github.com/NVIDIA-NeMo/Automodel/pull/3555/commits/46ba282f346403ed22b194745cf60e8c9b066344)

**Review by akoumpa — APPROVED** [review]
